crates/rattler_virtual_packages/src/lib.rs

```diff
@@ -34,16 +34,42 @@ pub mod linux;
 pub mod osx;
 
 use archspec::cpu::Microarchitecture;
+use core::fmt;
 use once_cell::sync::OnceCell;
 use rattler_conda_types::{GenericVirtualPackage, PackageName, Platform, Version};
+use std::any::type_name;
+use std::env;
 use std::hash::{Hash, Hasher};
+use std::str::FromStr;
 use std::sync::Arc;
 
 use crate::osx::ParseOsxVersionError;
 use libc::DetectLibCError;
 use linux::ParseLinuxVersionError;
 use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
+pub trait EnvOverride: Sized {
+    // Used to override the mechanisms here
+    fn from_env_var_name(env_var_name: &str) -> Option<Self>;
+
+    fn default_env_name() -> String;
+
+    fn from_default_env_var() -> Option<Self> {
+        Self::from_env_var_name(&Self::default_env_name())
+    }
+}
+
+impl<T: From<Version>> EnvOverride for T {
+    fn from_env_var_name(env_var_name: &str) -> Option<Self> {
+        let var = env::var(env_var_name).ok()?;
+        Some(Self::from(Version::from_str(&var).ok()?))
+    }
+
+    fn default_env_name() -> String {
+        format!("CONDA_{}_OVERIDE", type_name::<T>().to_uppercase())
+    }
```

Comment:
I dont think relying on type names is good here. I prefer just reimplementing this per type instead of using a blanket implementation

Reply:
The name should also be [`CONDA_OVERRIDE_CUDA`](https://github.com/conda/conda/blob/77aa3d337963fdbe5c817304a2052ae0125fd326/docs/source/user-guide/tasks/manage-virtual.rst#overriding-detected-packages)

Reply:
nice catch. The blanket implementation is opt in, is it really a problem?